lib/src/generate/build_impl.dart

```diff
@@ -294,3 +305,155 @@ class BuildImpl {
     }
   }
 }
+
+/// Extends [BuildImpl] to add the [runWatch] method, which watches all
+/// inputs for changes.
+class WatchImpl extends BuildImpl {
+  /// Injectable factory for creating directory watchers.
+  final DirectoryWatcherFactory _directoryWatcherFactory;
+
+  /// Delay to wait for more file watcher events.
+  final Duration _debounceDelay;
+
+  /// Whether or not we are currently watching and running builds.
+  bool _runningWatch = false;
+
+  /// Controller for the results stream.
+  StreamController<BuildResult> _resultStreamController;
+
+  /// All file listeners currently set up.
+  final _allListeners = <StreamSubscription>[];
```

Comment:
nit: This is personal preference, but I like putting a class's `final` properties above its reassignable ones.